agent/src/main/java/org/wso2/testgrid/agent/AgentApplication.java

```diff
@@ -0,0 +1,132 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.agent;
+
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.agent.beans.OperationRequest;
+import org.wso2.testgrid.agent.beans.OperationResponse;
+import org.wso2.testgrid.agent.listners.OperationResponseListener;
+import org.wso2.testgrid.agent.websocket.ClientEndpoint;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import javax.websocket.CloseReason;
+
+/**
+ * Agent bootstrap class.
+ *
+ * @since 1.0.0
+ */
+public class AgentApplication implements OperationResponseListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentApplication.class);
+    private static final String AGENT_PROP_FILE = File.separator + "opt" + File.separator + "testgrid" +
+            File.separator + "agent-config.properties";
+
+    private ClientEndpoint clientEndPoint;
+
+    public static void main(String args[]) {
+        new AgentApplication().startService();
+        while (true) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                break;
+            }
+        }
+    }
+
+    private void startService() {
+        String wsEndpoint;
+        String agentId;
+
+        Properties prop = new Properties();
+        InputStream input = null;
+        try {
+            if (new File(AGENT_PROP_FILE).exists()) {
+                input = new FileInputStream(AGENT_PROP_FILE);
+                prop.load(input);
+                wsEndpoint = prop.getProperty("wsEndpoint");
+                //aws:us-west-1:6718f976-7150-316d-9ec6-db29b7a2675f:i-0658922f45d25856f
+                agentId = prop.getProperty("provider") + ":" + prop.getProperty("region") + ":" +
+                        prop.getProperty("testPlanId") + ":" + prop.getProperty("instanceId");
+            } else {
+                logger.warn("Agent configurations not found.");
+                return;
+            }
+        } catch (IOException ex) {
+            logger.error("Error occurred when reading prop file: " + ex.getMessage(), ex);
+            return;
+        } finally {
+            if (input != null) {
+                try {
+                    input.close();
+                } catch (IOException e) {
+                    logger.error("Error occurred when closing prop file stream: " + e.getMessage(), e);
+                }
+            }
+        }
+
+        try {
+            logger.info("Agent ID: " + agentId);
+            // open web socket
+            clientEndPoint = new ClientEndpoint(new URI(wsEndpoint + agentId));
```

Comment:
This assumes the wsEndpoint will end with a '/'.

Reply:
Handled endpoint with or without '/' with https://github.com/wso2/testgrid/pull/747/commits/b2812b0818ed38457361934f505c931e587efcad